torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -320,7 +324,8 @@ def post_backward(self, *unused: Any):
                     fsdp_params_with_grad.append(fsdp_param)
                     unsharded_grads.append(fsdp_param.unsharded_grad_data)
                     fsdp_param.unsharded_param.grad = None
-            self.reshard()
+            if self.reshard_after_backward:
```

Comment:
However, it is also easy for us to support the other case: _do_ reduce gradients, but do not reshard parameters.

If people want to play around with different collective schedules during gradient accumulation, then this provides an extra lever.